util/annotations/annotations.go

```diff
@@ -27,7 +27,7 @@ import (
 // Each individual annotation is modeled by a Go error.
 // They are deduplicated based on the string returned by error.Error().
 // The zero value is usable without further initialization, see New().
-type Annotations map[string]error
+type Annotations map[string]annoErr
```

Comment:
Maybe we can leave this as is and then use type assertions and `error.As` as needed to do the right thing?

It would be weird to expose `annoErr` as an unexported type as part of this exported type.